imx: add kernel 6.12 support

Gateworks opened the pull request:
Allow selecting 6.12 as testing kernel on imx.

Runtime-tested:
 - imx8mm Gateworks Venice boards
 - imx8mp Gateworks Venice boards
 - imx6q/dl Gateworks Ventana boards

@Leo-PL can you please test on your i.MX7 board(s)?

Gateworks committed:
imx: add 6.12 kernel as testing

Allow selecting 6.12 as testing kernel on imx.

Signed-off-by: Tim Harvey <tharvey@gateworks.com>
Link: https://github.com/openwrt/openwrt/pull/19029
Signed-off-by: Nick Hainke <vincent@systemli.org>

Gateworks committed:
kernel: modules: video: fix video-coda for 6.12

Linux 6.8 changed the directory structure for the video-code driver.

Signed-off-by: Tim Harvey <tharvey@gateworks.com>
Link: https://github.com/openwrt/openwrt/pull/19029
Signed-off-by: Nick Hainke <vincent@systemli.org>

jonasjelonek reviewed (approved):
run-tested on USB armory Mk II (i.MX6ULZ, Cortex-A7). Support for this device is currently WIP in my tree. Everything I got working with 6.6 also works with 6.12 with this PR.

PolynomialDivision commented:
Why not using the kernel_bump script? https://github.com/openwrt/openwrt/blob/main/scripts/kernel_bump.sh

Gateworks committed:
kernel/imx: Create kernel files for v6.12 (from v6.6)

This is an automatically generated commit.

When doing `git bisect`, consider `git bisect --skip`.

Signed-off-by: Tim Harvey <tharvey@gateworks.com>
Link: https://github.com/openwrt/openwrt/pull/19029
Signed-off-by: Nick Hainke <vincent@systemli.org>

Gateworks committed:
kernel/imx: Restore kernel files for v6.6

This is an automatically generated commit which aids following Kernel patch
history, as git will see the move and copy as a rename thus defeating the
purpose.

For the original discussion see:
https://lists.openwrt.org/pipermail/openwrt-devel/2023-October/041673.html

Signed-off-by: Tim Harvey <tharvey@gateworks.com>
Link: https://github.com/openwrt/openwrt/pull/19029
Signed-off-by: Nick Hainke <vincent@systemli.org>

jonasjelonek commented:
> one commit that renames all the patches and a 2nd commit that adds them back for the old kernel? That's braindead I'm sorry to say. 

Afaik the reason for this is that git can keep track of history of these across kernel version bumps, due to the renaming. Recreating the files for older kernel version then is just necessary but they will be deleted anyway when the newer kernel becomes default.
See a more detailed reasoning in 3561015efd

PolynomialDivision commented:
Could you maybe split the copy of the old config and patches from the adaptation and changes for the new kernel? That would make the review easier and make it clearer what was changed for the new target.

Gateworks committed:
imx: 6-12: refresh patches and kernel configs

patches:
 - remove patches from 6.7-6.12 that are now upstream.
 - refresh remaining patches
 - 502-6.13-arm64-dts-freescale-rename-gw7905-to-gw75xx.patch was
   misnamed as that patch went upstream in 6.12 and thus is also removed
 - 504-6.13-arm64-dts-imx-Add-i.MX8M-Plus-Gateworks-GW82XX-2X-support.patch
   was refreshed to the final version that was accepted upstream
 - 600-PCI-imx6-Start-link-at-max-gen-first-for-IMX8MM-and-IMX8MP.patch
   was removed while I investigate an upstream approach for the issue
   it was working around.

configs:
 - config-6.12: unset new configs not needed for all cortexa7/a9/a53
 - cortexa53/config-default: added new CONFIG_PCI_IMX6_HOST config for cortexA53 (IMX8M)

Signed-off-by: Tim Harvey <tharvey@gateworks.com>
Link: https://github.com/openwrt/openwrt/pull/19029
Signed-off-by: Nick Hainke <vincent@systemli.org>

Gateworks commented:
> Could you maybe split the copy of the old config and patches from the adaptation and changes for the new kernel? That would make the review easier and make it clearer what was changed for the new target.

Sure - I pushed a new series which uses the kernel bump script with a refresh patch on top that isolates the changes

PolynomialDivision commented:
Thanks! Rebased on top of main and merged!
